src/index.js

```diff
@@ -188,8 +194,15 @@ class Libp2p extends EventEmitter {
     log('libp2p is stopping')
 
     try {
-      this.pubsub && await this.pubsub.stop()
-      this._dht && await this._dht.stop()
+      await Promise.all([
+        this.pubsub && this.pubsub.stop(),
+        this._dht && this._dht.stop()
+      ])
+
+      for (const dial of this.dialer.pendingDials.values()) {
```

Comment:
I would add a `destroy` function to `this.dialer` that does this and rename to `this.dialer._pendingDials`. This is state that does not need to be public or shared - right?